CONTRIBUTING.md

```diff
@@ -0,0 +1,92 @@
+# 🤝 Contributing
+
+Thank you for your interest in contributing to Tabby! We appreciate all contributions. For a better experience and support, join us on [Slack](https://links.tabbyml.com/join-slack)!
+
+## Local Setup
+
+To begin contributing to Tabby, first clone the repository locally:
+
+```
+git clone --recurse-submodules https://github.com/TabbyML/tabby
+```
+
+If you have already cloned the repository, you can initialize submodules with this command:
+
+```
+git submodule update --recursive --init
+```
+
+Make sure you have installed [Rust](https://www.rust-lang.org/learn/get-started), and the following dependencies are for MacOS or Debian-based Linux distributions:
+
+```bash
+# For MacOS
+brew install protobuf
+
+# For Ubuntu / Debian
+apt-get install protobuf-compiler libopenblas-dev
+```
+
+Before proceeding, ensure that all tests are passing locally:
+
+```
+cargo test -- --skip golden
```

Comment:
add some explaination on why we wanna exclude golden in non-mac m1/m2 systems.
(btw, for m1/m2, it's still fine to run golden test)